### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by Macarse**

### CODE REVIEW

File: contracts/Strategy.sol [hunk 1]
```diff
@@ -6,42 +6,29 @@ pragma solidity 0.6.12;
 pragma experimental ABIEncoderV2;
 
 // These are the core Yearn libraries
-import {
-    BaseStrategy,
-    StrategyParams
```
Comment (Macarse) [inline contracts/Strategy.sol:11]: You were not using this

File: contracts/Strategy.sol [hunk 2]
```diff
@@ -6,42 +6,29 @@ pragma solidity 0.6.12;
...
-} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {BaseStrategy} from "@yearnvaults/contracts/BaseStrategy.sol";
+import {SafeERC20, SafeMath, IERC20, Address} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import "@openzeppelin/contracts/math/Math.sol";
 
-
 // Import interfaces for many popular DeFi projects, or add your own!
 import "../interfaces/I1INCHGovernance.sol";
 import "../interfaces/I1INCHGovernanceRewards.sol";
-interface I1nchStake is I1INCHGovernance, IERC20 {}
```
Comment (Macarse) [inline contracts/Strategy.sol:25]: I killed `I1nchStake` insce it's not an `IERC20`.

File: contracts/Strategy.sol [hunk 3]
```diff
@@ -6,42 +6,29 @@ pragma solidity 0.6.12;
...
-    IERC20 public OneInchToken = IERC20(0x111111111117dC0aa78b770fA6A738034120C302);
     //Initiate 1inch interfaces
-    I1nchStake public stakeT = I1nchStake(0xA0446D8804611944F1B527eCD37d7dcbE442caba);
+    I1INCHGovernance public stakeT = I1INCHGovernance(0xA0446D8804611944F1B527eCD37d7dcbE442caba);
     I1INCHGovernanceRewards public governanceT = I1INCHGovernanceRewards(0x0F85A912448279111694F4Ba4F85dC641c54b594);
-    // The amount of 1INCH we have earned which we are now staking in 1INCH Governance
-    uint256 public staking = 0;
 
     constructor(address _vault) public BaseStrategy(_vault) {
-        require(address(want) == address(OneInchToken),"wrong want token");
```
Comment (Macarse) [inline contracts/Strategy.sol:39]: it didn't make sense to me to have a variable in storage just to check that the constructor is doing things right. Up to you.

File: contracts/Strategy.sol [hunk 4]
```diff
@@ -79,31 +66,31 @@ contract Strategy is BaseStrategy {
...
         _profit = balanceOfWant().sub(balanceOfWantBefore);
     }
 
     function adjustPosition(uint256 _debtOutstanding) internal override {
-        // Invest the rest of the want
         uint256 _wantAvailable = balanceOfWant();
-        if (_wantAvailable > 0) {
-            stakeT.stake(_wantAvailable);
+
+        if (_debtOutstanding >= _wantAvailable) {
```
Comment (Macarse) [inline contracts/Strategy.sol:82]: If we have `want` to return, no reason to stake.

File: interfaces/I1INCHGovernance.sol [hunk 5]
```diff
@@ -3,4 +3,5 @@ pragma solidity 0.6.12;
 interface I1INCHGovernance {
     function stake(uint256 amount) external;
     function unstake(uint256 amount) external;
-}
\ No newline at end of file
+    function balanceOf(address account) external view returns (uint256);
```
Comment (Macarse) [inline interfaces/I1INCHGovernance.sol:6]: this is the only ERC20 like function you are using.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    "devDependencies": {
+        "@commitlint/cli": "^11.0.0",
+        "@commitlint/config-conventional": "^11.0.0",
+        "ethlint": "^1.2.5",
+        "prettier": "^2.2.1",
+        "prettier-plugin-solidity": "^1.0.0-beta.1",
+        "pretty-quick": "^3.0.2"
+    },
+    "scripts": {
+        "prettier": "prettier --write 'contracts/**/*.sol'"
```
Comment (Macarse) [inline package.json:12]: you can lint solidity code with `yarn run prettier`

File: tests/conftest.py [hunk 7]
```diff
@@ -1,54 +1,25 @@
 import pytest
```
Comment (Macarse) [inline tests/conftest.py:1]: Make sure you add a test to lower debt ratio and increase debt ratio.
here you have an example:
https://github.com/Macarse/yhegic/blob/master/tests/development/test_debt_limit.py#L29

Comment (akshaynexus) [reply tests/conftest.py:1]: Will do

**Review by akshaynexus — APPROVED** [review]

**Pull request merged**